### ISSUE TITLE
Improvement to maintenance mode add/or check alert disable

### ISSUE THREAD

**Issue by P6g9YHK6** [amidaware/tacticalrmm#1770]
**Is your feature request related to a problem? Please describe.**
it is currently impossible to disable individual checks without changing the whole policy affected to the agent.
it is also impossible to plan duration for maintenances or make them limited in time.

**Describe the solution you'd like**
_Granular Alert Suppression:_
We would like the ability to selectively suppress specific types of alerts during Maintenance Mode(only check/tasks/or overdue). 
Also the ability to target specific check to disable during the maintenance.
This granular would allow us to tailor the feature according to our precise requirements, ensuring that only relevant alerts are muted while critical notifications remain active. 

_Flexible Maintenance Window:_
It would be good to have the option to define customizable maintenance windows. To allow us to specify the start and end times for Maintenance Mode based on our schedule. This flexibility ensures that alerts are muted only during the designated maintenance period.

_Notification History:_
Access to a log of suppressed alerts during Maintenance Mode.

_Rambling:_
Maybe a simple right click with a button "downtime" that asks for the set time of the DT on each individual check/tasks/agent would be the best way to handle this without too much effort and could replace all together the maintenance mode feature ?

**Describe alternatives you've considered**
no alernative found

**Additional context**
https://assets.nagios.com/downloads/nagioscore/docs/nagioscore/3/en/downtime.html

**Comment by datawolk** [comment]
Would be a valuable feature!

**Comment by P6g9YHK6** [comment]
linking this :
https://github.com/amidaware/tacticalrmm/issues/1064
as it could be well implemented inside this framework

**Comment by LPJon** [comment]
It would also be nice to allow an Admin of the TacticalRMM Server to disable any task or check even when it is part of an automation policy for any specific device. Otherwise you have to completely deactivate the policy in order to stop the task/check from being performed. This will affect many systems which wouldn't need the policy disabled for troubleshooting.

**Comment by P6g9YHK6** [comment]
> It would also be nice to allow an Admin of the TacticalRMM Server to disable any task or check even when it is part of an automation policy for any specific device. Otherwise you have to completely deactivate the policy in order to stop the task/check from being performed. This will affect many systems which wouldn't need the policy disabled for troubleshooting.

Yes same as you can "mass run" a task from the policy page you could "mass disable" or "mass downtime" a single check/task
while agent overdue would need a new button on that page to set a downtime or disable on the alert for that policy
